core/trino-main/src/test/java/io/trino/cost/TestComparisonStatsCalculator.java

```diff
@@ -695,6 +708,237 @@ public void symbolToCastExpressionNotEqual()
+                        .lowValue(0)
+                        .highValue(20)
+                        .distinctValuesCount(30)
+                        .nullsFraction(0))
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
+
+        rowCount = nonNullRowCount * (overlappingFractionX * OVERLAPPING_RANGE_INEQUALITY_FILTER_COEFFICIENT);
+        assertCalculate(new ComparisonExpression(GREATER_THAN, new SymbolReference("x"), new SymbolReference("w")))
+                .outputRowsCount(rowCount)
+                .symbolStats("x", symbolAssert -> symbolAssert.averageRowSize(4)
+                        .lowValue(0)
+                        .highValue(10)
+                        .distinctValuesCount(20)
+                        .nullsFraction(0))
+                .symbolStats("w", symbolAssert -> symbolAssert.averageRowSize(8)
+                        .lowValue(0)
+                        .highValue(10)
+                        .distinctValuesCount(15)
+                        .nullsFraction(0))
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
+
+        // Flip symbols to be on opposite sides
+        double overlappingFractionW = 0.5;
+        rowCount = nonNullRowCount * (overlappingFractionW * OVERLAPPING_RANGE_INEQUALITY_FILTER_COEFFICIENT);
+        assertCalculate(new ComparisonExpression(LESS_THAN, new SymbolReference("w"), new SymbolReference("x")))
+                .outputRowsCount(rowCount)
+                .symbolStats("x", symbolAssert -> symbolAssert.averageRowSize(4)
+                        .lowValue(0)
+                        .highValue(10)
+                        .distinctValuesCount(20)
+                        .nullsFraction(0))
+                .symbolStats("w", symbolAssert -> symbolAssert.averageRowSize(8)
+                        .lowValue(0)
+                        .highValue(10)
+                        .distinctValuesCount(15)
+                        .nullsFraction(0))
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
+
+        double alwaysGreaterFractionW = 0.5;
+        rowCount = nonNullRowCount * (alwaysGreaterFractionW + overlappingFractionW * OVERLAPPING_RANGE_INEQUALITY_FILTER_COEFFICIENT);
+        assertCalculate(new ComparisonExpression(GREATER_THAN, new SymbolReference("w"), new SymbolReference("x")))
+                .outputRowsCount(rowCount)
+                .symbolStats("x", symbolAssert -> symbolAssert.averageRowSize(4)
+                        .lowValue(-10)
+                        .highValue(10)
+                        .distinctValuesCount(40)
+                        .nullsFraction(0))
+                .symbolStats("w", symbolAssert -> symbolAssert.averageRowSize(8)
+                        .lowValue(0)
+                        .highValue(20)
+                        .distinctValuesCount(30)
+                        .nullsFraction(0))
+                .symbolStats("z", equalTo(capNDV(zStats, rowCount)));
+
+        // Open ranges
```

Comment:
do we have case for `unknowns` (not open ranges)?

Reply:
we have a case with `unknownRange` and `unknownNdvRange`